tests/integration/project/cmd_project_test.go

```diff
@@ -38,6 +39,40 @@ var _ = Describe("odo project command tests", func() {
 		})
 	})
 
+	Context("Should be able to delete a project with --wait", func() {
+		var projectName string
+		JustBeforeEach(func() {
+			projectName = helper.RandString(6)
+		})
+
+		It("--wait should work with deleting a project", func() {
+
+			// Create the project
+			helper.CmdShouldPass("odo", "project", "create", projectName)
+
+			// Delete with --wait
+			output := helper.CmdShouldPass("odo", "project", "delete", projectName, "-f", "--wait")
+			Expect(output).To(ContainSubstring("Waiting for project to be deleted"))
```

Comment:
We should also list the projects and test that the project deletion was successful.

Reply:
+1. We should check if this worked as expected.

Reply:
@cdrage Maybe wait for the required conditions to be met using `WaitForCmdOut`

Reply:
@mik-dass @dharmit Unfortunately I don't think we can with the whole project deletion being asynchronous...